### PR TITLE
Milestone 2 redux react

### PR DESCRIPTION
**Pull request by mikenath223**
Implementations:
 * Create components, containers, actions, reducers directories.
*  Implement the App, Book, BookForm, BookList components.
* Define the redux action and reducers.
* Initialize the redux store with each book in the initial state having a unique ID
* Rendered the <App/> component wrapped in <Provider> and pass the store down to the components.

Kindly review. Thanks.

### CODE REVIEW

**Comment by bellom** [comment]
> Implementations:
> 
> * Create components, containers, actions, reducers directories.
> * Implement the App, Book, BookForm, BookList components.
> * Define the redux action and reducers.
> * Initialize the redux store with each book in the initial state having a unique ID
> * Rendered the  component wrapped in  and pass the store down to the components.
> 
> Kindly review. Thanks.

I will advise @mikenath223 to commit your changes whenever you add new features, look at @ViriCruz  commit history to yours...
Microverse support equal or close number of commits from both students...
image

File: src/index.js [hunk 1]
```diff
@@ -1,16 +1,31 @@
...
 
-ReactDOM.render(<App/>, document.getElementById('root'))
\ No newline at end of file
+const store = createStore(rootReducer);
+const Container = connect(mapStateToProps, mapDispatchToProps)(App);
+
+ReactDOM.render(
+  <Provider store={store}>
+    <Container />
+  </Provider>,
```
Comment (bellom) [inline src/index.js]: I am curious, `const Container` doesn't look like a react component you could render in `Provider` tag.

File: src/index.js [hunk 2]
```diff
@@ -1,16 +1,31 @@
...
 
-  render() {
-    return (
-      <div>
-        <h1>Hello World!</h1>
-      </div>
-    )
-  }
-}
+const mapDispatchToProps = dispatch => ({
```
Comment (bellom) [inline src/index.js]: Guys, we should have this in `Booklist` component, to be on a safer side and to buy ourself some time why can't we just respect and follow the Microverse project guidelines as the rest of the students 🙏 
thanks.
**state object seems missing in here, kindly check the instructions below from the pathright**
image
https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/44896827/

File: src/components/app.js [hunk 3]
```diff
@@ -0,0 +1,12 @@
+import React from 'react';
+import BooksList from '../containers/booksList';
+import BooksForm from '../containers/booksForm';
+
+const App = () => (
+  <main>
```
Comment (bellom) [inline src/components/app.js]: Let's change this to a `div` tag.
Let's respect and follow the Microverse project guidelines as the rest of the students. 🙏
image

File: src/containers/booksList.js [hunk 4]
```diff
@@ -0,0 +1,26 @@
+import React from 'react';
+import Book from '../components/book';
+
```
Comment (bellom) [inline src/containers/booksList.js:6]: Let's move our `mapStateToProp` in `index.js` down here.
image

File: src/actions/index.js [hunk 5]
```diff
@@ -0,0 +1,16 @@
+
+const ADD_BOOK = 'addBook';
+const DEL_BOOK = 'delBook';
```
Comment (bellom) [inline src/actions/index.js:3]: I'm curious about these line 2 and 3, do you really need them in here?
And CREATE_BOOK action objects seems not completed.

example:
```
const addBook = book => ({
  type: 'CREATE_BOOK',
  id: book.id,
  title: book.title,
  category: book.category,
});


const deleteBook = book => ({
  type: 'DELETE_BOOK',
  id: book,
});
```

File: src/reducers/book.js [hunk 6]
```diff
@@ -0,0 +1,38 @@
+import { ADD_BOOK, DEL_BOOK } from '../actions/index';
+
+const defaultState = [
```
Comment (bellom) [inline src/reducers/book.js]: Our book object should be in `src/index.js` right?
image

**Review by bellom — CHANGES_REQUESTED** [review]
## Hi Team :clap: 

### Awesome job so far!! :+1: 

#### Let try as much as possible to follow the pathwright instructions 👍 
 
**Kindly check out my comments below;**

**Kindly re-submit for review after making all those changes :slightly_smiling_face: Thumps -up await you.....**

**Best,**

**Oluwaseun Bello**
**_Slack Handle - @sean bellom_**

File: src/index.js (hunk 1 above)
Comment (mikenath223) [reply src/index.js]: Actually wanted to create a container component that would be directly connected to redux, unlike a presentational component that isn't directly connected to redux. However, we will change it to a presentational UI component. Thanks

File: src/components/app.js (hunk 3 above)
Comment (mikenath223) [reply src/components/app.js]: Thanks. The change has been made.

File: src/index.js (hunk 2 above)
Comment (mikenath223) [reply src/index.js]: All done.

Comment (ViriCruz) [reply src/index.js]: sorry for this, we found instructions of pathwright a little hard to follow, we tried to reproduce freecodecamp and egghead.io tutorials into this project to fulfill the requirements, we are doing our best. I'm so sorry for the headache we caused you but thanks for your nice feedback 💪

File: src/actions/index.js (hunk 5 above)
Comment (ViriCruz) [reply src/actions/index.js:3]: We are using destructuring object here. 
`const CREATE_BOOK = book => ({
  type: ADD_BOOK,
  book
});`

This is the same as above: 
`const CREATE_BOOK = book => ({
  type: ADD_BOOK,
  book: {
     bookId: 1,
     title: 'Name of book',
     category: 'Action'
  },
});`

I don't know if this answers your doubts. 😄

**Review by Oghenebrume50 — APPROVED** [review]
### STATUS: APPROVED 

Hi @mikenath223 @ViriCruz good job :+1:  everything looks clean and nice please merge and move on to the next milestone :slightly_smiling_face: **This is approved**

**RAPHAEL NORIODE** :world_map:
